python-lib/sharepoint_lists.py

```diff
@@ -24,27 +24,14 @@ def get_sharepoint_type(dss_type):
     return DSSConstants.TYPES.get(dss_type, SharePointConstants.FALLBACK_TYPE)
 
 
-def matched_item(column_ids, column_names, item, column_to_expand=None):
-    ret = {}
-    for key, value in item.items():
+def column_ids_to_names(column_ids, column_names, sharepoint_row):
+    """ Replace the column ID used by SharePoint by their column names for use in DSS"""
+    dss_row = {}
+    for key, value in sharepoint_row.items():
         if key in column_ids:
             name = column_names[key]
-            ret[name] = value
-    return ret
-
-
-def expand_matched_item(column_ids, column_names, item, column_to_expand=None):
-    ret = {}
-    column_to_expand = {} if column_to_expand is None else column_to_expand
-    for key, value in item.items():
-        if key in column_ids:
-            name = column_names[key]
-            key_to_return = column_to_expand.get(key)
-            if key_to_return:
-                ret[name] = value.get(key_to_return)
-            else:
-                ret[name] = value
-    return ret
+            dss_row[name] = value
+    return dss_row
```

Comment:
More concise and pythonic way to do the same thing. I find it more understandable to have everything in the dict but it is subjective.
```python
return {column_names[key]: value for key, value in sharepoint_row.items() if key in column_ids}
```